mock/mock.go

```diff
@@ -701,7 +702,9 @@ func (args Arguments) Diff(objects []interface{}) (string, int) {
 
 			// normal checking
 
-			if assert.ObjectsAreEqual(expected, Anything) || assert.ObjectsAreEqual(actual, Anything) || assert.ObjectsAreEqual(actual, expected) {
+			expectedAnythingAndRequired := assert.ObjectsAreEqual(expected, Anything) && actual != missing
```

Comment:
One suggestion would be to move away from usage of the string, you could change the missing to `var missingArgument = errors.New("(Missing)")`, so we will be comparing on error values, and its easy to change output if required. like this function could return error (custom type) and based on it, the output should be formatted.